app/models/effects/external.rb

```diff
@@ -0,0 +1,48 @@
+module Effects
+  class External < Effect
+    class ExternalEffectFailed < StandardError; end
+
+    def perform(workflow_id, subject_id)
+      reductions = SubjectReduction.where(
+        workflow_id: workflow_id, 
+        subject_id: subject_id, 
+      )
+      reductions = reductions.where(reducer_key: config[:reducer_key]) if config[:reducer_key]
+
+      begin
+        if valid?
```

Comment:
This check could be moved to the top of the method. `raise InvalidConfiguration unless valid?`

Reply:
✔️